packages/core/core/src/BundleGraph.js

```diff
@@ -1092,32 +1092,45 @@ export default class BundleGraph {
         ? this._graph.getNodeIdByContentKey(startAsset.id)
         : bundleNodeId,
       getChildren: nodeId => {
-        let children = this._graph
-          .getNodeIdsConnectedFrom(nodeId)
-          .map(id => [id, nullthrows(this._graph.getNode(id))]);
-
-        let sorted =
-          entries && bundle.entryAssetIds.length > 0
-            ? children.sort(([, a], [, b]) => {
-                let aIndex = bundle.entryAssetIds.indexOf(a.id);
-                let bIndex = bundle.entryAssetIds.indexOf(b.id);
-
-                if (aIndex === bIndex) {
-                  // If both don't exist in the entry asset list, or
-                  // otherwise have the same index.
-                  return 0;
-                } else if (aIndex === -1) {
-                  return 1;
-                } else if (bIndex === -1) {
-                  return -1;
-                }
-
-                return aIndex - bIndex;
-              })
-            : children;
-
-        entries = false;
-        return sorted.map(([id]) => id);
+        let children = this._graph.getNodeIdsConnectedFrom(nodeId);
+
+        if (entries && bundle.entryAssetIds.length > 0) {
+          entries = false;
+          return children.sort((a, b) => {
+            let aNode = nullthrows(this._graph.getNode(a));
+            let bNode = nullthrows(this._graph.getNode(b));
+
+            let aIndex = bundle.entryAssetIds.indexOf(aNode.id);
+            let bIndex = bundle.entryAssetIds.indexOf(bNode.id);
+
+            if (aIndex === bIndex) {
+              // If both don't exist in the entry asset list, or
+              // otherwise have the same index.
+              return 0;
+            } else if (aIndex === -1) {
+              return 1;
+            } else if (bIndex === -1) {
+              return -1;
+            }
+
+            return aIndex - bIndex;
+          });
+        }
+
+        let node = nullthrows(this._graph.getNode(nodeId));
+        if (node.type === 'asset') {
+          let deps = [...node.value.dependencies.keys()];
```

Comment:
I had kinda hoped to get rid of dependencies being stored in two places at some point. But idk how else to do this unless we make graph edges ordered. WDYT @lettertwo @wbinnssmith?

Reply:
I think something else may already fix this, because we just encountered an issue caused by `reverse()` in `getReferencedBundle()`  for the experimental bundler (which caused script tags to be inserted in a different order based on bundler, which the test was indexing), but once we removed that, default bundler was fine (i.e. passing entire test suite). 

So do we know that the order is currently not already dependency order? 🤔

Reply:
It usually is dependency order, but this PR is about this order being nondeterministic. I think your situation is deterministic. So this still needs to be fixed separately

Reply:
Edges are appended on add (and spliced on remove), so there is an inherent order already, but it is independent of type. Maybe we could add a way to insert an edge (rather than append), though what the API might look like is a bit unobvious to me.  Maybe something like `graph.insertEdgeAfter(from, to, type, after)` that looks up the edge for `(from, to, type)` and then splices in a new edge for `(from, after, type)`...

Reply:
> I had kinda hoped to get rid of dependencies being stored in two places at some point.

One approach is making that dependency an array of ids to still keep the order: https://github.com/parcel-bundler/parcel/pull/8286